PythonScripts/UniFolderCreator/UniFolderCreator.py

```diff
@@ -1,29 +1,48 @@
 import os
 from sys import exit
+from pathlib import Path
 
 # Directory path to wherever you have your folder for the semester.
 # Eg 'C:\Users\Tim\Google Drive\2023 Sem 2'
-folder_dir = r"E:\QUT\2023 Semester 1"
-semester_weeks = 15 # Modify this if needed
+FOLDER_DIR: Path = Path.cwd()
+SEMESTER_WEEKS: int = 12 # Modify this if needed
+DRY_RUN: bool = True
 
-# Given a path to a folder, returns all folders within.
-def get_subdirectories(directory):
-    return [f.name for f in os.scandir(folder_dir) if f.is_dir()]
-    
-# Create a subfolder for each week of the semester for each subject folder.
-subs = get_subdirectories(folder_dir)
-if subs == []:
-    print("Create the main folders for your respective classes before running script!")
-    exit()
-    
-for sub in subs:
-    for week in range(1, semester_weeks+1):
-        new_folder = f"Week {week}"
-        full_path = f"{folder_dir}\\{sub}\\{new_folder}"
-        print(f"Creating folder at {full_path}")
-        if os.path.exists(full_path):
-            print("Folder already exists!")
+def main() -> None:
+    # Create a subfolder for each week of the semester for each subject folder.
+    subs = get_subdirectories(FOLDER_DIR)
+
+    if len(subs) == 0:
+        print("ERROR:\tCreate the main folders for your respective classes before running script!")
+        exit()
+        
+    for sub in subs:
+        make_directories(sub, SEMESTER_WEEKS)
+
+def get_subdirectories(directory: Path) -> list[Path]:
+    '''Returns each top-level subdirectory in a folder.'''
+    return [f for f in directory.iterdir() if f.is_dir()]
+
+def make_directories(parent: Path, weeks: int) -> None:
+    '''Creates a folder for each week in the specified parent.'''
+    for week in range(1, weeks+1):
+        new_folder_name = f"Week {week}"
+        full_path: Path = parent.joinpath(new_folder_name)
+
+        if full_path.exists():
+            print(f"WARNING:\tFolder {full_path} already exists!")
             pass
```

Comment:
Why the random pass here?

Reply:
Whoops, yeah that doesn't need to be there. Must've left it in on accident

Reply:
also everything else looks great